### DISCUSSION TITLE
v0.2.2+15

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#633] [Announcements]
## Alpha Release [v0.2.2+15]

I know it's been a while since the last Thunder general release, but the wait is finally over! There have been many improvements, additions, changes and fixes in this release which should make for a even better Thunder experience.

A reminder for anyone who didn’t catch the previous announcements, Thunder is now officially available on Google Play and on the App Store! You should also be able to search up Thunder on the Play Store/App Store to find it. For more details about the launch, see the following posts: 
- Google Play Announcement: https://lemmy.world/post/1631968
- App Store Announcement: https://lemmy.world/post/2926034

If you are using the Google Play version or App Store version, please note that it may take a bit of time before you receive the update. As always, **this is an alpha release**, so do expect bugs to pop up here and there. Now onto the update notes. This update brings a few major features:

- Complete overhaul of the Floating Action Buttons (FAB) - contributions from @CTalvio and @micahmo 
  - You can swipe up from the FAB to show a list of extended actions to perform
  - You can customize your FAB actions within the new settings page for FABs
  - You can customize short and long press actions for the FAB in the feed/post views. Changing the short press action will change the default FAB action.
- Complete overhaul of the Create Post screen - contribution from @coslu 
  - All markdown text and preview icons are now at the bottom for better accessibility
  - Link previews show up when adding a new link when creating the post
- Inbox badges - you will now be able to see if you have any unread inbox items when you first launch Thunder, or when you switch accounts. This is an initial feature that will be improved upon in the future to give more frequent updates when you receive new notifications
- Overall improvements to the UI - contribution from @CTalvio, @micahmo, @tom-james-watson and others!
  - Improved post badges to indicate images, text, links better
  - Improvements to how read posts are shown in the feed for better contrast/readability
  - Improvements to performance when loading the feed and external image thumbnails
  - Domains now show up on the post to indicate where the link will take you

And much much more. I can't go through all the changes in the changelog, but for those who are curious, feel free to check out the full changelog below.

We have a Matrix space if you would like to join in on discussions: https://matrix.to/#/#thunderapp:matrix.org

If you find any issues or would like to suggest features, please open up a new issue! As always, contributions are always welcomed here.

### Added
- Added long press on profile icon to bring up profile modal @zdzoz
- Added spinning circle animation on comment card while waiting for comment to be deleted/restored - contribution from @ajsosa
- Added vote count to comment card in comment list for the user profile - contribution from @ajsosa
- Show instance taglines in the feed - contribution from @micahmo
- Added new visual swipe gesture picker within settings - contribution from @micahmo
- Added option to toggle tappable authors and communities in the feed - contribution from @micahmo
- Added spinner indicator when sharing media - contribution from @micahmo
- Added inbox unread indicators on the bottom navigation bar
- Added more robust community navigation - contribution from @micahmo
- Added support for navigating from image preview to comments - contribution from @micahmo
- Added inkwell effect to comments - contribution from @CTalvio
- Redesigned UI for creating posts - contribution from @coslu
- Added moderator identifier to comments - contribution from @micahmo
- Added ability to navigate to user's profile from comment body - contribution from @micahmo
- Added support for exact community name search - contribution from @micahmo
- Overhauled floating action button with expandable and customizable actions - contribution from @CTalvio
- Added additional localization strings to Thunder, and added temporary language files for Swedish/Finnish
- Added manual refreshing to the user account page - contribution from @micahmo 
- Added inkwell effect when tapping on usernames in comments - contribution from @micahmo 
- Added additional font scaling options for comments and metadata
- Long-pressing on FAB shows extended actions - contribution from @micahmo
- Added support for customizable short-press and long-press FAB actions - contribution from @micahmo
- Added thumbnail badges to posts for more clarity - contribution from @CTalvio
- Added domain for posts linking to external websites - contribution from @CTalvio
- Added comment navigation buttons - contribution from @micahmo
- Added full screen swipe to go back on main pages
- Added new option scrape missing external link previews which is off by default. Its purpose is to attempt to find an image when an external link thumbnail is not available - contribution @ajsosa

### Changed
- Removed tap zones for author/community on compact post cards - contribution from @CTalvio
- Creating, deleting, and restoring a comment will update locally without requiring a refetch - contribution from @ajsosa
- Added caching to images to improve overall experience - contribution from @micahmo
- Respect comment deleted in reply modal, and inbox - contribution from @ajsosa
- Improvements to sort picker to allow for navigating back when selecting top option - contribution from @micahmo
- Minor UI improvements to comment images, community image banners, and image viewer - contribution from @CTalvio
- Minor sidebar shadow adjustment - contribution from @CTalvio
- Snappier image load transition - contribution from @micahmo 
- Align back button in image preview with the back button in the main pages - contribution from @micahmo 
- Moved location of comment button within image preview - contribution from @micahmo 
- Adjusted font scaling to be platform specific
- Improve behavior of deferred comment indicator - contribution from @micahmo
- Text scaling now respects system's font scaling. Text scaling is based off of the system font
- Improved contrast on user chips and badges - contribution from @CTalvio 
- Show external link previews option is now scrape missing external link previews and off by default for performance reasons - contribution from @ajsosa
- Make it easier to distinguish different post types in the Compact List View - contribution from @tom-james-watson
- Improved the colour and contrast of read posts in the feed - contribution from @micahmo 

### Fixed
- Fixed issue where the community post feed was missing the last post - contribution from @ajsosa
- Fixed the gesture conflict that can occur between pinch to zoom and tap slide to zoom - contribution from @CTalvio
- Fixed incorrect indentation to load more replies card within comments - contribution from @ajsosa
- Fixed another edge case of the loading more comments infinite spinning circle - contribution from @ajsosa
- Fixed infinite spinning circle when loading a user's posts in the user profile - contribution from @ajsosa
- Fixed issue where toast notifications were not showing up in the post page - contribution from @ajsosa
- Removed sliver of border color that was present on root comments for both thick and thin style comments - contribution from @ajsosa
- Fixed issue where saving an image on Android would save to Pictures/Pictures/Thunder instead of Pictures/Thunder
- Fixed comment highlighting for comment context regression - contribution from @ajsosa 
- Fixed another instance of infinite spin for comment loading - contribution from @ajsosa 
- Fixed mis-aligned previews in comfort cards for edge-to-edge links from @Fmstrat
- Fixed missing community icons in feed - contribution from @sant0s12
- Fixed issue where more posts would not load if initial posts fit the screen
- Fixed issue where compact feed would not load properly when "Enable Link Preview" setting was turned on
- Fixed semantic issue where user comments would read the improper value for downvotes
- Fixed issue where you could not vote/save comments in quick succession
- Fix improper back button handling - contribution from @micahmo 
- Fixed feed page reaching the end in some cases where NSFW content is turned on
- Fixed issue where external link thumbnails weren't being displayed due to show external link previews option being off which was only intended to prevent html scraping - contribution from @ajsosa
- Reduce image viewer pan inertia - contribution from @CTalvio 

<hr /><em>This discussion was created from the release <a href='https://github.com/thunder-app/thunder/releases/tag/v0.2.2%2B15'>v0.2.2+15</a>.</em>

**Comment by kovalensky** [comment]
### The dude is a machine, real chad

![CSWhQ1JUYAAg8kl](https://github.com/thunder-app/thunder/assets/45331093/ece7cd75-a144-421e-863a-d6792f4f6986)

**Comment by Charlotte-br560** [comment]
Great work on the Alpha Release! Exciting to see all the improvements and new features. Keep up the fantastic work!

**Reply by kovalensky** [reply]
My asian bros are rocking! :D

**Reply by Fmstrat** [reply]
Crazy spam users updating READMEs and posting on old things to build rep.
